src/Uri.php

```diff
@@ -437,9 +437,9 @@ public function withScheme($scheme)
 
     public function withUserInfo($user, $password = null)
     {
-        $info = $user;
+        $info = rawurlencode($user);
         if ($password != '') {
```

Comment:
please change that to test against `null`. an empty string should be possible and add the seperator according to rfc 3986

Reply:
OK